atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -1484,6 +1486,32 @@ public void commitThrowsIfRolledBackAtCommitTime_expiredLocks() {
                 .hasLocksExpired(1);
     }
 
+    @Test
+    public void commitThrowsIfLockAcquisitionFails() {
+        final Cell cell = Cell.create(PtBytes.toBytes("row1"), PtBytes.toBytes("column1"));
+
+        TimelockService timelockServiceMock = mock(TimelockService.class);
+        // Let request to acquire commit locks time out
+        when(timelockServiceMock.lock(any(), any())).thenReturn(LockResponse.timedOut());
+
+        ConjureStartTransactionsResponse conjureResponse = startTransactionWithWatches();
+        LockImmutableTimestampResponse res = conjureResponse.getImmutableTimestamp();
+        long transactionTs = conjureResponse.getTimestamps().start();
+
+        Transaction snapshot = getSnapshotTransactionWith(timelockServiceMock, () -> transactionTs, res, unused -> {});
+
+        // Need a writer to trigger writer commit protocol
+        snapshot.put(TABLE, ImmutableMap.of(cell, PtBytes.toBytes("value")));
+
+        // Lock acquisition should fail and be marked as time out
```

Comment:
I believe the assert itself explains what's written above, but let me know if you disagree :)
We also generally prefer assertThatThrownBy, for new code (again, there's a mix in this class), with an assert on the message itself (See getSweepSentinelUnderCommittedWriteThrowsAndCanBeRetried)